### DISCUSSION TITLE
Override error with MIDI-Input example. VSC & PlatformIO

### DISCUSSION THREAD

**Discussion by MiOiM** [tttapa/Control-Surface#641] [Q&A]
Hi @tttapa, thank you for your amazing work! 

I am having a problem compiling the MIDI-Input example (or the other two examples related to MIDI callbacks). I am getting the following error:
```
void MyMIDI_Callbacks::onChannelMessage(CS::MIDI_Interface&, CS::ChannelMessage)' marked 'override', but does 
not override
```

I think it could be related to Visual Studio Code or PlatformIo. When I use the Arduino IDE I don't get the error, but I use VSC on all my other projects and would like to use it on this one too.
 
The Control Surface library is correctly installed in PlatformIo, and I have successfully compiled other examples which work as expected, but as soon as I add the **struct MyMIDI_Callbacks : MIDI_Callbacks** section (copy-pasted to my working sketch, or by itself on a new project) I get this error. 

Do I need to add any other header to the code, or something like that?

Thanks a lot in advance!

**Comment by tttapa** [comment]
It sounds like you're using a new tutorial with an older version of the library. Try installing the latest `master` version from GitHub, or follow the documentation for the specific version you have installed, see https://tttapa.github.io/Control-Surface-doc/.

IIRC, you can install the latest version using 
```
pio lib install https://github.com/tttapa/Control-Surface.git#master
```
